be/src/exec/vectorized/hdfs_scan_node.cpp

```diff
@@ -565,28 +576,36 @@ bool HdfsScanNode::_filter_partition(const std::vector<ExprContext*>& partition_
 }
 
 Status HdfsScanNode::_find_and_insert_hdfs_file(const THdfsScanRange& scan_range) {
-    if (_partition_values_map.find(scan_range.partition_id) == _partition_values_map.end()) {
+    if (_iceberg_table == NULL && (_partition_values_map.find(scan_range.partition_id) == _partition_values_map.end())) {
         // partition has been filtered
         return Status::OK();
     }
 
+    std::string scan_range_path = scan_range.full_path;
+    if (_hdfs_table != NULL) {
+        scan_range_path = scan_range.relative_path;
+    }
+
     // search file in hdfs file array
     // if found, add file splits to hdfs file desc
     // if not found, create
     for (auto& item : _hdfs_files) {
-        if (item->partition_id == scan_range.partition_id && item->path == scan_range.relative_path) {
+        if (item->partition_id == scan_range.partition_id && item->path == scan_range_path) {
             item->splits.emplace_back(&scan_range);
             return Status::OK();
         }
     }
 
-    auto* partition_desc = _hdfs_table->get_partition(scan_range.partition_id);
+    std::string& native_file_path = const_cast<string&>(scan_range.full_path);
+    if (_hdfs_table != NULL) {
+        auto* partition_desc = _hdfs_table->get_partition(scan_range.partition_id);
 
-    SCOPED_TIMER(_open_file_timer);
+        SCOPED_TIMER(_open_file_timer);
 
-    std::filesystem::path file_path(partition_desc->location());
-    file_path /= scan_range.relative_path;
-    const std::string& native_file_path = file_path.native();
+        std::filesystem::path file_path(partition_desc->location());
+        file_path /= scan_range.relative_path;
+        native_file_path = file_path.native();
```

Comment:
Because native_file_path is a reference, this will change scan_range.full_path. Is it your intention.
If it is, I don't think it is a good idea to change an input argument.

Reply:
I change native_file_path to a variable without reference. If we need use reference for performance consideration i will refactor later. PTAL @imay thanks